src/porepy/models/geometry.py

```diff
@@ -36,24 +36,69 @@ def set_geometry(self) -> None:
         self.set_fracture_network()
         self.set_md_grid()
         self.nd: int = self.mdg.dim_max()
+
         # If fractures are present, it is advised to call
         pp.contact_conditions.set_projections(self.mdg)
 
+        self.set_well_network()
+        if len(self.well_network.wells) > 0:
+            # Compute intersections
+            assert isinstance(self.fracture_network, pp.FractureNetwork3d)
+            pp.compute_well_fracture_intersections(
+                self.well_network, self.fracture_network
+            )
+            # Mesh fractures and add fracture + intersection grids to mixed-dimensional
+            # grid along with these grids' new interfaces to fractures.
+            self.well_network.mesh(self.mdg)
+
     def set_fracture_network(self) -> None:
         """Assign fracture network class."""
         self.fracture_network = pp.FractureNetwork2d()
 
+    def set_well_network(self) -> None:
+        """Assign well network class."""
+        self.well_network = pp.WellNetwork3d()
+
+    def is_well(self, grid: pp.Grid | pp.MortarGrid) -> bool:
+        """Check if a subdomain is a well.
+
+        Parameters:
+            sd: Subdomain to check.
+
+        Returns:
+            True if the subdomain is a well, False otherwise.
+
+        """
+        if isinstance(grid, pp.Grid):
+            return getattr(grid, "well_num", -1) >= 0
```

Comment:
We should document the Grid attribute well_num (and frac_num) at some point, but let's keep it hidden for now, I can see this undergo changes as we learn more about how we want to do things.